### PR TITLE
Do not create instance role when `skipDefaultNodeGroup` is enabled

### PR DESCRIPTION
**Pull request by flostadler**
When enabling `skipDefaultNodeGroup` we should not be creating resources related to the default node group. This includes the default instance role.

With this change the `Cluster` component will no longer create this default role when `skipDefaultNodeGroup` is enabled.

One of the tests needs re-recording because we're introducing a breaking change (i.e. not creating the extraneous instance role), but in doing so we're aligning the provider with user expectations. Once this is merged in I'll release the next version and re-record.

Fixes https://github.com/pulumi/pulumi-eks/issues/1176

### CODE REVIEW

**Comment by flostadler** [comment]
The change is failing the `TestEksAuthModeUpgrade` test that has a recorded baseline. This is expected because we're changing the default behavior of the cluster here, but we're more closely aligning it with customer expectations.

**Comment by flostadler** [comment]
This is a bit more complex than initially expected. The node groups use the cluster's default instanceProfile when none is passed in. Therefore they also use the default instance role.

The better course of action here is to add a flag like `skipDefaultInstanceRole`.

File: sdk/python/pulumi_eks/_inputs.py [hunk 1]
```diff
@@ -289,7 +289,7 @@ class ClusterNodeGroupOptionsArgsDict(TypedDict):
 
         Note: Given the inheritance of auto-generated CF tags and `cloudFormationTags`, you should either supply the tag in `autoScalingGroupTags` or `cloudFormationTags`, but not both.
         """
-        bootstrap_extra_args: NotRequired[str]
+        bootstrap_extra_args: NotRequired[pulumi.Input[str]]
```
Comment (flostadler) [inline sdk/python/pulumi_eks/_inputs.py:292]: Merging https://github.com/pulumi/pulumi-eks/pull/1414 & https://github.com/pulumi/pulumi-eks/pull/1415 ontop of each other dropped those changes.
This adds is back now, work tree is clean again

File: docs/eks-v3-migration.md [hunk 2]
```diff
@@ -160,3 +160,7 @@ The `NodeGroup` and `NodeGroupV2` components now accept inputs for the following
 - `nodeAssociatePublicIpAddress`
 
 If you're using Go you'll need to adjust your program to handle those types being inputs.
+
+### Cluster does not create extraneous instance IAM role if `skipDefaultNodeGroup` is set to `true`
+
+Previously the Cluster component created a default instance IAM role even if `skipDefaultNodeGroup` was set to `true`. This role gets correctly omitted now if you're specifying `skipDefaultNodeGroup`.
```
Comment (t0yv0) [inline docs/eks-v3-migration.md:166]: "if you are"

File: sdk/python/pulumi_eks/_inputs.py (hunk 1 above)
Comment (t0yv0) [reply sdk/python/pulumi_eks/_inputs.py:292]: Are we missing CI check guarantees that the worktree should be clean possibly?

**Review by t0yv0 — APPROVED** [review]

File: sdk/python/pulumi_eks/_inputs.py (hunk 1 above)
Comment (flostadler) [reply sdk/python/pulumi_eks/_inputs.py:292]: We're merging those into a feature branch that's only built when cutting an alpha release or when triggered manually. It's checking it as part of those workflows

**Pull request merged**